test/linalg/test_linalg_solve.fypp

```diff
@@ -0,0 +1,190 @@
+
+        type(linalg_state_type) :: state
+
+        ${rt}$ :: A(3,3) = transpose(reshape([${rt}$ :: 1,-1, 2, &
+                                                            0, 1, 1, &
+                                                            1,-1, 3], [3,3]))
+        ${rt}$ :: b(3,3) = transpose(reshape([${rt}$ :: 0, 1, 2, &
+                                                            1,-2,-1, &
+                                                            2, 3,-1], [3,3]))
+        ${rt}$ :: res(3,3) = transpose(reshape([${rt}$ ::-5,-7,10, &
+                                                           -1,-4, 2, &
+                                                            2, 2,-3], [3,3]))
+        ${rt}$ :: x(3,3)
+
+        x = solve(a,b,err=state)
+        
+        call check(error,state%ok(),state%print())
+        if (allocated(error)) return
+        
+        call check(error, all(abs(x-res)<abs(res*epsilon(0.0_${rk}$))), 'results match expected')
+        if (allocated(error)) return        
+        
+    end subroutine test_${ri}$_solve_multiple
+    #:endif
+    #:endfor
+
+    #:for rk,rt,ri in CMPLX_KINDS_TYPES
+    #:if rk!="xdp"
+    !> Complex linear system
+    !> Militaru, Popa, "On the numerical solving of complex linear systems",
+    !> Int J Pure Appl Math 76(1), 113-122, 2012.
+    subroutine test_${ri}$_solve(error)
+        type(error_type), allocatable, intent(out) :: error
+
+        type(linalg_state_type) :: state
+
+        ${rt}$ :: A(5,5), b(5), res(5), x(5)
+        integer(ilp) :: i
+
+        ! Fill in linear system
+        A = (0.0_${rk}$,0.0_${rk}$)
+
+        A(1:2,1) = [(19.73_${rk}$,0.0_${rk}$),(0.0_${rk}$,-0.51_${rk}$)]
+        A(1:3,2) = [(12.11_${rk}$,-1.0_${rk}$),(32.3_${rk}$,7.0_${rk}$),(0.0_${rk}$,-0.51_${rk}$)]
+        A(1:4,3) = [(0.0_${rk}$,5.0_${rk}$),(23.07_${rk}$,0.0_${rk}$),(70.0_${rk}$,7.3_${rk}$),(1.0_${rk}$,1.1_${rk}$)]
+        A(2:5,4) = [(0.0_${rk}$,1.0_${rk}$),(3.95_${rk}$,0.0_${rk}$),(50.17_${rk}$,0.0_${rk}$),(0.0_${rk}$,-9.351_${rk}$)]
+        A(3:5,5) = [(19.0_${rk}$,31.83_${rk}$),(45.51_${rk}$,0.0_${rk}$),(55.0_${rk}$,0.0_${rk}$)]
+
+        b = [(77.38_${rk}$,8.82_${rk}$),(157.48_${rk}$,19.8_${rk}$),(1175.62_${rk}$,20.69_${rk}$),(912.12_${rk}$,-801.75_${rk}$),(550.0_${rk}$,-1060.4_${rk}$)]
+
+        ! Exact result
+        res = [(3.3_${rk}$,-1.0_${rk}$),(1.0_${rk}$,0.17_${rk}$),(5.5_${rk}$,0.0_${rk}$),(9.0_${rk}$,0.0_${rk}$),(10.0_${rk}$,-17.75_${rk}$)]
+
+        x = solve(a,b,err=state)
+
+        call check(error,state%ok(),state%print())
+        if (allocated(error)) return
+        
+        call check(error, all(abs(x-res)<abs(res)*1.0e-3_${rk}$), 'results match expected')
+        if (allocated(error)) return        
+
+    end subroutine test_${ri}$_solve
+
+    !> 2x2 Complex linear system
+    !> https://math.stackexchange.com/questions/1996540/solving-linear-equation-systems-with-complex-coefficients-and-variables
+    subroutine test_2x2_${ri}$_solve(error)
```

Comment:
Is the license of this content compatible with the MIT license? It should be checked before merging this PR.

Reply:
https://opensource.stackexchange.com/questions/3966/can-i-legally-use-code-i-found-on-a-forum#:~:text=In%20the%20case%20of%20all,manner%20any%20changes%20you%20make.

> If it's not simple and you can't/don't want to rewrite it, then you need to consider how it's licensed on the website or forum you found it on. In the case of all StackExchange sites, scroll to the bottom where you'll see a link to [CC BY-SA 3.0](https://creativecommons.org/licenses/by-sa/3.0/). You are free to use and change any content (code or text) that you find on SE sites, as long as you give credit and as long as you continue to share in the same manner any changes you make.

Seems like it is compatible.

Reply:
This [post](https://opensource.stackexchange.com/a/7436) seems to suggest an answer a bit different. 

@perazz Could you consider a different reference that would avoid this potential license issue?

Reply:
Sounds good @jvdp1 @jalvesz, I will create a brand new complex test system, so all license issues are avoided. Apologies, I did not think a linear system could be covered by copyright.

Reply:
Test removed and replaced at [4458b88](https://github.com/fortran-lang/stdlib/pull/806/commits/4458b88248c2c6fb72f91e0843a98f445d852bb2)